File: typescript/Basic/OAuth/src/oauth.ts
import { ApiClient, FrequenciesApi, CurrenciesApi } from '@factset/sdk-paengine';
import { ConfidentialClient } from '@factset/sdk-utils';

async function oauth() {
    const client = new ApiClient();
    client.factsetOauth2Client = new ConfidentialClient("../../../app-config.json");

    const frequenciesApi = new FrequenciesApi(client);
    const result = await frequenciesApi.getPAFrequencies();

    console.log(result.data);
}

oauth();

File: dotnet/Basic/OAuth/OAuth.cs
using System;
using FactSet.SDK.Utils.Authentication;
using FactSet.SDK.PAEngine.Api;
using FactSet.SDK.PAEngine.Model;
using System.Threading;
using System.Threading.Tasks;
using System.Collections.Generic;

namespace Factset.SDK.Sample.Basic.Oauth
{
    class Oauth
    {
        static readonly CancellationTokenSource cts = new CancellationTokenSource();

        static async Task Main(string[] args)
        {
            FactSet.SDK.PAEngine.Client.Configuration configuration = new FactSet.SDK.PAEngine.Client.Configuration();
            ConfidentialClient confidentialClient = await ConfidentialClient.CreateAsync("../../../app-config.json");
            configuration.OAuth2Client = confidentialClient;

            FrequenciesApi frequenciesApi = new FrequenciesApi(configuration);

            try
            {
                cts.CancelAfter(5000);

                FrequencyRoot frequencyRoot = await frequenciesApi.GetPAFrequenciesAsync(cts.Token);

                foreach (var pair in frequencyRoot.Data)
                {
                    Console.WriteLine($"{pair.Key} : {pair.Value}");
                }
            }
            catch (OperationCanceledException)
            {
                Console.WriteLine("\nTasks cancelled: timed out.\n");
            }
            finally
            {
                cts.Dispose();
            }
        }
    }
}
